src/main/java/seedu/rc4hdb/model/StringField.java

```diff
@@ -23,11 +23,23 @@ public boolean containsIgnoreCase(String substring) {
     }
 
     /**
-     * Returns true if {@code other} is a subclass of {@code this} or if {@code this} is a subclass of {@code other}.
+     * Returns true if {@code other} and {@code this} comply to the sub or super class relation and their {@code value}
+     * must be the same, ignoring case (upper or lower).
      * @return true if the condition above is satisfied.
      */
-    private boolean isSubclass(StringField other) {
-        return other.getClass().equals(this.getClass());
+    public boolean equalsIgnoreCase(StringField other) {
+        return other == this
+                || (this.isSubOrSuperClass(other)
+                && this.value.equalsIgnoreCase(other.value));
+    }
+
+    /**
+     * Returns true if {@code other} is a subclass of {@code this} or if {@code this} is a subclass of {@code other}
+     * during runtime.
+     * @return true if the condition above is satisfied.
+     */
+    private boolean isSubOrSuperClass(StringField other) {
+        return other != null && other.getClass().equals(this.getClass());
     }
```

Comment:
We might want to rewrite these using guard clauses as per the code quality guidelines in the textbook.